keps/sig-storage/3141-prevent-volume-mode-conversion/README.md

```diff
@@ -338,29 +340,69 @@ decisions.
 
 ### Test Plan
 
-E2E tests will be added for this design, that attempt to restore a volume with
-and without requisite privileges. 
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
 
-#### Unit tests
+##### Prerequisite testing updates
 
-- With feature flag disabled:
-  - attempt to convert volume mode when creating a `PVC`
-  from a `VolumeSnapshot`.
-- With feature flag enabled, attempt to convert volume mode when creating a `PVC`
-from a `VolumeSnapshot`:
-  - With `Spec.SourceVolumeMode` populated and `snapshot.storage.kubernetes.io/allowVolumeModeChange: true`
-  annotation present.
-  - With `Spec.SourceVolumeMode` populated but no `snapshot.storage.kubernetes.io/allowVolumeModeChange: true`
-  annotation.
-  - With `Spec.SourceVolumeMode` set to `nil`.
+<!--
+ Based on reviewers feedback describe what additional tests need to be added prior
+ implementing this enhancement to ensure the enhancements have also solid foundations.
+ -->
+
+None. New E2E tests will be added for the transition to beta.
+
+##### Unit tests
+
+<!--
+ In principle every added code should have complete unit test coverage, so providing
+ the exact set of tests will not bring additional value.
+ However, if complete unit test coverage is not possible, explain the reason of it
+ together with explanation why this is acceptable.
+ -->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+The unit tests were added to the CSI external-provisioner repo. 
+
+- https://github.com/kubernetes-csi/external-provisioner/pull/726/
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- No integration tests added.
 
-#### E2E tests
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
```

Comment:
Please add a link to the added e2e test and also a link to show the testgrid.

Reply:
Done